doc/cdi-populators.md

```diff
@@ -0,0 +1,123 @@
+Example for an instance of VolumeImportSource CRD:
+
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeImportSource
+metadata:
+  name: my-import-source
+spec:
+  source:
+      http:
+         url: "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
+```
+
+### Using populators with PVCs
+User can create a CR and PVCs specifying the CR in the `DataSourceRef` field and those will be handled by the matching populator controller.
+
+#### Import
+Example for PVC which use the VolumeImportSource above that will be handled by the import populator:
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name:  my-pvc
+spec:
+  dataSourceRef:
+    apiGroup: cdi.kubevirt.io
+    kind: VolumeImportSource
+    name: my-import-source
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+```
+
+Above PVC will trigger the reconcile of the import populator controller.
+The controller will create a matching temporary PVC with the appropriate annotations, which will get bound and populated.
+Once the temporary PVC population is done, the PV will be rebound to the original PVC completing the population process.
+
+#### Upload
+Example of VolumeUploadSource and a PVC that will be handled by the upload populator:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeUploadSource
+metadata:
+  name: my-upload-source
+spec:
+   contentType: kubevirt
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name:  my-pvc
+spec:
+  dataSourceRef:
+    apiGroup: cdi.kubevirt.io
+    kind: VolumeUploadSource
+    name: my-upload-source
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+```
+After creating the VolumeUploadSource and PVC, you can start the upload to the pvc as describe in the [upload doc](upload.md).
+
+#### Clone
+Example of VolumeCloneSource and a PVC that will be handled by the clone populator:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: VolumeCloneSource
+metadata:
+  name: my-clone-source
+spec:
+  source:
+    kind: PersistentVolumeClaim
+    name: my-pvc-source
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name:  my-cloned-pvc
+spec:
+  dataSourceRef:
+    apiGroup: cdi.kubevirt.io
+    kind: VolumeCloneSource
+    name: my-clone-source
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+
+```
+
+### Using populators with DataVolumes
+
+The integration of datavolumes and CDI populators is seamless. You can create the datavolumes the same way you always have.
+If the used storage class is CSI storage then the datavolume population will occur via the CDI populators with the end result as always has been of a populated PVC. You will be able to notice that the created PVC will stay pending until the population process completes.
```

Comment:
If the DataVolume targets a storage class that uses a CSI provisioner CDI will automatically use the new populators method.  The behavior will be the same as always but with the following key differences.  <different Pending DV status, and PVC will not become bound until is is populated>.